lib/ravelin/event.rb

```diff
@@ -75,13 +75,11 @@ def payload_customer_reference_present?
       self.payload.keys.any? {|k| %i(customer_id temp_customer_id).include? k }
     end
 
-    def convert_to_epoch(time)
-      if time.is_a?(Time)
-        time.to_i
-      elsif time.is_a?(DateTime) && time.respond_to?(:to_i)
-        time.to_i
-      elsif time.is_a?(Integer)
-        time
+    def convert_to_epoch(val)
+      if val.is_a?(Time) || val.is_a?(Date) || val.is_a?(DateTime)
```

Comment:
Do you need this complexity? It seems like `datetime_to_epoch` will cope with integers, dates, times and datetimes already?